src/altinn-app-frontend/src/shared/resources/dataLists/fetchDataListsSaga.ts

```diff
@@ -0,0 +1,133 @@
+  IRuntimeState,
+} from 'src/types';
+
+import { get } from 'altinn-shared/utils';
+
+export const formLayoutSelector = (state: IRuntimeState): ILayouts | null => state.formLayout?.layouts;
+export const formDataSelector = (state: IRuntimeState) => state.formData.formData;
+export const dataListsSelector = (state: IRuntimeState): IDataLists => state.dataListState.dataLists;
+export const dataListsWithIndexIndicatorsSelector = (state: IRuntimeState) =>
+  state.dataListState.dataListsWithIndexIndicator;
+export const instanceIdSelector = (state: IRuntimeState): string | undefined => state.instanceData.instance?.id;
+export const repeatingGroupsSelector = (state: IRuntimeState) => state.formLayout?.uiConfig.repeatingGroups;
+
+export function* fetchDataListsSaga(): SagaIterator {
+  const layouts: ILayouts = yield selectNotNull(formLayoutSelector);
+  const repeatingGroups: IRepeatingGroups = yield selectNotNull(repeatingGroupsSelector);
+  const fetchedDataLists: string[] = [];
+  const dataListsWithIndexIndicators: IDataListsMetaData[] = [];
+  for (const layoutId of Object.keys(layouts)) {
+    for (const element of layouts[layoutId] || []) {
+      if (element.type !== 'List' || !element.dataListId) {
+        continue;
+      }
+
+      const { dataListId, mapping, secure } = element;
+
+      const { keys, keyWithIndexIndicator } = getDataListLookupKeys({
+        id: dataListId,
+        mapping,
+        secure,
+        repeatingGroups,
+      });
+      if (keyWithIndexIndicator) {
+        dataListsWithIndexIndicators.push(keyWithIndexIndicator);
+      }
+
+      if (!keys?.length) {
+        continue;
+      }
+
+      for (const dataListsObject of keys) {
+        const { id, mapping, secure } = dataListsObject;
+        const lookupKey = getDataListLookupKey({ id, mapping });
+        if (dataListId && !fetchedDataLists.includes(lookupKey)) {
+          yield fork(fetchSpecificDataListSaga, {
+            dataListId,
+            dataMapping: mapping,
+            secure,
+          });
+          fetchedDataLists.push(lookupKey);
+        }
+      }
+    }
+  }
+  yield put(
+    dataListsActions.setDataListsWithIndexIndicators({
+      dataListsWithIndexIndicators,
+    }),
+  );
+}
+
+export function* fetchSpecificDataListSaga({
+  dataListId,
+  dataMapping,
+  secure,
+}: IFetchSpecificDataListSaga): SagaIterator {
+  const key = getDataListLookupKey({ id: dataListId, mapping: dataMapping });
+  const instanceId = yield select(instanceIdSelector);
+  try {
+    const metaData: IDataListsMetaData = {
+      id: dataListId,
+      mapping: dataMapping,
+      secure,
+    };
+    yield put(dataListsActions.fetching({ key, metaData }));
+    const formData: IFormData = yield select(formDataSelector);
+    const language = yield select(appLanguageStateSelector);
+    const dataList = yield select(listStateSelector);
+
+    const pageSize = dataList.dataLists[dataListId].size ? dataList.dataLists[dataListId].size.toString() : '5';
```

Comment:
Does this mean the size is 5 items by default for all data lists, regardless of the component setup? :thinking: Isn't that number configurable using the `pagination` property? And if it's not set, I guess all items should be fetched, not just 5.